# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBIndex.cpp
# Commit: 23fc7469065d
# Full Hash: 23fc7469065d32effcad818083aa3840c95f26d5
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-11 21:53:06
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Reduce code duplication for different open cursor variants. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40956
# ==============================================================================

diff -r 0e77abe58616 -r 23fc7469065d dom/indexedDB/IDBIndex.cpp
--- a/dom/indexedDB/IDBIndex.cpp	Wed Sep 11 11:51:26 2019 +0000
+++ b/dom/indexedDB/IDBIndex.cpp	Wed Sep 11 11:51:24 2019 +0000
@@ -455,28 +455,12 @@
 
   IDBCursor::Direction direction = IDBCursor::ConvertDirection(aDirection);
 
-  OpenCursorParams params;
-  if (aKeysOnly) {
-    IndexOpenKeyCursorParams openParams;
-    openParams.commonIndexParams().commonParams().objectStoreId() =
-        objectStoreId;
-    openParams.commonIndexParams().indexId() = indexId;
-    openParams.commonIndexParams().commonParams().optionalKeyRange() =
-        std::move(optionalKeyRange);
-    openParams.commonIndexParams().commonParams().direction() = direction;
+  const CommonIndexOpenCursorParams commonIndexParams = {
+      {objectStoreId, std::move(optionalKeyRange), direction}, indexId};
 
-    params = std::move(openParams);
-  } else {
-    IndexOpenCursorParams openParams;
-    openParams.commonIndexParams().commonParams().objectStoreId() =
-        objectStoreId;
-    openParams.commonIndexParams().indexId() = indexId;
-    openParams.commonIndexParams().commonParams().optionalKeyRange() =
-        std::move(optionalKeyRange);
-    openParams.commonIndexParams().commonParams().direction() = direction;
-
-    params = std::move(openParams);
-  }
+  const auto params =
+      aKeysOnly ? OpenCursorParams{IndexOpenKeyCursorParams{commonIndexParams}}
+                : OpenCursorParams{IndexOpenCursorParams{commonIndexParams}};
 
   RefPtr<IDBRequest> request = GenerateRequest(aCx, this);
   MOZ_ASSERT(request);